taf/TAF_Metadata.py

```diff
@@ -1024,6 +977,44 @@ def getFormatsForValidationAndRecode():
 
     vs_transp_CCS_Cat = ["239"]
 
+    @staticmethod
+    def getFormatsForValidationAndRecode():
+        """
+        Helper function to get formats for validation and recode.
+        """
+        from pyspark.sql.functions import col, to_date
+
+        spark = SparkSession.getActiveSession()
+
+        subData = [
+            ("35", "ELG", "CSO", "2018-04-01", "9999-12-01"),
+            ("35", "TPL", "CSO", "2018-04-01", "9999-12-01"),
+            ("48", "ELG", "CSO", "2023-04-05", "9999-12-01"),
+        ]
+
+        subSchema = StructType(
+            [
+                StructField("submtg_state_cd", StringType(), True),
+                StructField("fil_type", StringType(), True),
+                StructField("submsn_type", StringType(), True),
+                StructField("submsn_eff_dt_string", StringType(), True),
+                StructField("submsn_end_dt_string", StringType(), True),
+            ]
+        )
+
+        if spark is not None:
+            subDF = (
+                spark.createDataFrame(data=subData, schema=subSchema)
+                .withColumn(
+                    "submsn_eff_dt", to_date(col("submsn_eff_dt_string"), "yyyy-MM-dd")
+                )
+                .withColumn(
+                    "submsn_end_dt", to_date(col("submsn_end_dt_string"), "yyyy-MM-dd")
+                )
+            )
+
+            subDF.createOrReplaceTempView("state_submsn_type")
```

Comment:
we can also chain these two together without declaring subDF

Reply:
These are fantastic questions! So far, in our stages, we've been tasked with converting the SAS code over to Python. We've been literally doing a 1 to 1 conversion first to get that running. You're more than likely right that much of this can be optimized. I'll take a look at your suggestions, but we've been hesitant to change anything until we get TAF running in Python and away from the SAS system. Let me know if you have more questions and I'll explain more :-)